src/mbgl/renderer/sources/render_raster_dem_source.cpp

```diff
@@ -26,18 +27,21 @@ void RenderRasterDEMSource::update(Immutable<style::Source::Impl> baseImpl_,
 
     enabled = needsRendering;
 
-    optional<Tileset> _tileset = impl().getTileset();
-
-    if (tileset != _tileset) {
-        tileset = _tileset;
+    const optional<Tileset>& implTileset = impl().tileset;
+    // In Continuous mode, keep the existing tiles if the new tileset is not
+    // yet available, thus providing smart style transitions without flickering.
```

Comment:
What would happen if new tileset is never updated (source stopped providing data). Would we use stale tileset indefinitely?

Reply:
It can happen only on `Source::loadDescription` failure, that will end up to `MapObserver::onDidFailLoadingMap()` error notification.